### PR TITLE
Benchmarks

### PR DESCRIPTION
**Pull request by rubinovitz**
@smothers here it is, back from the dead. Happy Halloween! 💀
If it looks good: I can merge it in if I'm online or if you want to do the manual merge feel free.

### CODE REVIEW

**Comment by rubinovitz** [comment]
I think if we do merge it in we should shut off the benchmarking action until we're ready.

**Comment by smothers** [comment]
@rubinovitz I agree. I'd like to be able to use the checks for CD so we should turn off checks we don't expect to pass

**Comment by rubinovitz** [comment]
Should I just remove the workflow for now before we merge?

File: shadow-cljs.edn [hunk 1]
```diff
@@ -18,4 +18,10 @@
                            :output-feature-set :es6}
         :modules {:main {:init-fn example.core/init}}
         :js-options {:resolve {"devcards-marked" {:target :npm :require "marked"}
-                               "devcards-syntax-highlighter" {:target :npm :require "highlight.js"}}}}}}
+                               "devcards-syntax-highlighter" {:target :npm :require "highlight.js"}}}}
+  :npm-dist {:target :npm-module
```
Comment (smothers) [inline shadow-cljs.edn:22]: Is this related to this benchmarks. If not please submit it as a separate PR. I know it's a pain, but I'd like to get in the habit of making PRs be about just 1 thing. This makes them easier to review and also makes it more reasonable to enforce squash merges since the final commit message will have a single theme and be revertable with no side effects.

File: package.json [hunk 2]
```diff
@@ -6,7 +6,7 @@
   "homepage": "https://github.com/homebaseio/homebase-react",
   "repository": {
     "type": "git",
-    "url": "https://github.com/homebaseio/homebase-react.git"
+    "url": "git+https://github.com/homebaseio/homebase-react.git"
```
Comment (smothers) [inline package.json:9]: Is this related to benchmarks?

File: package.json [hunk 3]
```diff
@@ -15,16 +15,30 @@
     "state",
     "graph"
   ],
+  "main": "./dist/homebase.react.js",
```
Comment (smothers) [inline package.json:18]: I think this is also part of a separate packaging PR

File: package.json [hunk 4]
```diff
@@ -15,16 +15,30 @@
...
-    "react-dom": "16.13.0",
+    "react": "16.14.0",
+    "react-component-benchmark": "0.0.4",
+    "react-dom": "16.14.0",
     "shadow-cljs": "2.11.4"
   },
-  "dependencies": {}
+  "dependencies": {},
+  "bugs": {
+    "url": "https://github.com/homebaseio/homebase-react/issues"
```
Comment (smothers) [inline package.json:38]: and this

**Comment by smothers** [comment]
@rubinovitz given we're not ready to turn the benchmarks on yet what do you think about?
1. pulling the packaging code into a separate PR
1. significantly relaxing the timings on the benchmarks so they pass and can be used for CI
    1. Or: pulling the JS testing code into a separate PR with some simplified passing tests just to kick off CI and revisiting benchmarks later

**Comment by rubinovitz** [comment]
Yeah there’s some of this that are resolving conflicts that emerged when a bunch was merged in while this branch was broken. I can go back and go through it

File: package.json (hunk 3 above)
Comment (rubinovitz) [reply package.json:18]: I think this is actually a bad conflict resolution from the hacky merge with master I had to do so looking into it

File: package.json (hunk 4 above)
Comment (rubinovitz) [reply package.json:38]: happy to just kill this

File: package.json (hunk 2 above)
Comment (rubinovitz) [reply package.json:9]: no I'll remove

File: shadow-cljs.edn (hunk 1 above)
Comment (rubinovitz) [reply shadow-cljs.edn:22]: this is formatting I can remove

**Comment by rubinovitz** [comment]
@smothers The packaging code is a result of me having to manually merge a bunch of code and accidentally modifying the packing code. I'll go back and remove the packing code changes.

**Comment by rubinovitz** [comment]
> @rubinovitz given we're not ready to turn the benchmarks on yet what do you think about?
> 
> 1. pulling the packaging code into a separate PR
> 2. significantly relaxing the timings on the benchmarks so they pass and can be used for CI
>    
>    1. Or: pulling the JS testing code into a separate PR with some simplified passing tests just to kick off CI and revisiting benchmarks later


#1 is just going to occur when I remove the packaging changes the arose during the messy merge, #2 is fine

**Comment by rubinovitz** [comment]
@rubinovitz reduce number of components to 100

**Review by smothers — APPROVED** [review]
